### PR TITLE
docs(package.json): add docs for the `resolutions` block

### PR DESCRIPTION
**Pull request by IgorMinar**
Closes yarnpkg/yarn#4754

### CODE REVIEW

**Comment by Haroenv** [comment]
Deploy preview ready!

Built with commit 5491d06b81ef73a84a28fb7c5a56db5f571b20d2

https://deploy-preview-687--yarnpkg.netlify.com

File: lang/en/docs/package-json.md [hunk 1]
```diff
@@ -363,6 +363,21 @@ If your package only allows one version of a given dependency, and you'd like to
...
+{
+  "resolutions": {
+    "transitive-package-1": "0.0.29",
+    "transitive-package-2": "file:./local-forks/transitive-package-2",
+    "dependencies-package-1/transitive-package-3": "^2.1.1"
+  }
+}
+```
+
+Allows you to override a version of a particular nested dependency. See [the Selective Versions Resolutions RFC](https://github.com/yarnpkg/rfcs/blob/master/accepted/0000-selective-versions-resolutions.md) for the full spec.
```
Comment (kaylieEB) [inline lang/en/docs/package-json.md]: This link will be changed soon the be under /implemented folder once this PR is merged https://github.com/yarnpkg/rfcs/pull/85

File: lang/en/docs/package-json.md [hunk 2]
```diff
@@ -363,6 +363,21 @@ If your package only allows one version of a given dependency, and you'd like to
 
 Note that if your `package.json` contains `"flat": true` and other packages depend on yours (e.g. you are building a library rather than an application), those other packages will also need `"flat": true` in their `package.json` or be installed with `yarn install --flat` on the command line.
```
Comment (kaylieEB) [inline lang/en/docs/package-json.md]: It may also be worth adding that running yarn install with `--flat` will auto-add resolutions for you

**Comment by Haroenv** [comment]
Thanks!

**Comment by kaylieEB** [comment]
@Haroenv can we merge this? I don't have rights on this repo :P

**Comment by Haroenv** [comment]
Thanks for pinging me @kaylieEB

**Comment by kaylieEB** [comment]
np, thanks @Haroenv !

**Pull request merged**